lib/ts/recipe/emailpassword/components/features/emailVerification/emailVerification.tsx

```diff
@@ -0,0 +1,296 @@
+        // If props provided by user, and successfully handled.
+        if (this.props.onHandleSuccess !== undefined) {
+            await this.props.onHandleSuccess(context);
+        }
+
+        // Otherwise, do nothing.
+    };
+
+    onSignInClicked = (): void => {
+        const onSuccessRedirectURL = this.getRecipeInstanceOrThrow().getConfig().signInAndUpFeature
+            .onSuccessRedirectURL;
+        redirectToInApp(onSuccessRedirectURL, undefined, this.props.history);
+    };
+
+    onCallVerifyEmailAPI = async (requestJson: RequestJson, headers: HeadersInit): Promise<VerifyEmailAPIResponse> => {
+        // If props provided by user.
+        if (this.props.onCallVerifyEmailAPI !== undefined) {
+            return this.props.onCallVerifyEmailAPI(requestJson, headers);
+        }
+
+        // Otherwise, use default.
+        return this.getRecipeInstanceOrThrow().verifyEmailAPI(requestJson, headers);
+    };
+
+    onCallSendVerifyEmailAPI = async (headers: HeadersInit): Promise<SendVerificationEmailAPIResponse> => {
+        // If props provided by user.
+        if (this.props.onCallSendVerifyEmailAPI !== undefined) {
+            return this.props.onCallSendVerifyEmailAPI(headers);
+        }
+
+        // Otherwise, use default.
+        return this.getRecipeInstanceOrThrow().sendVerificationEmailAPI(headers);
+    };
+
+    signOut = async (): Promise<void> => {
+        // If props provided by user.
+        if (this.props.signOut !== undefined) {
+            await this.props.signOut();
+            return;
+        }
+
+        // Otherwise, use default.
+        try {
+            await signOut();
+            redirectToInApp(
+                this.getRecipeInstanceOrThrow()
+                    .getAppInfo()
+                    .websiteBasePath.getAsStringDangerous(),
+                undefined,
+                this.props.history
+            );
+        } catch (e) {}
+        return;
+    };
+
+    getSessionRecipe(): Session | undefined {
+        return SuperTokens.getDefaultSessionRecipe();
+    }
+
+    doesSessionExist = async (): Promise<boolean> => {
+        // If props provided by user.
+        if (this.props.doesSessionExist !== undefined) {
+            return await this.props.doesSessionExist();
+        }
+
+        const sessionRecipe = this.getSessionRecipe();
+        if (sessionRecipe !== undefined) {
+            return sessionRecipe.doesSessionExist();
+        }
+
+        // Otherwise, return false.
+        return false;
+    };
+
+    redirectToVerifyEmailScreen = async (): Promise<void> => {
+        redirectToInApp(
+            `${this.getRecipeInstanceOrThrow()
+                .getAppInfo()
+                .websiteBasePath.getAsStringDangerous()}${DEFAULT_VERIFY_EMAIL_PATH}?rid=emailpassword`,
+            undefined,
+            undefined
+        ); // No history object provided here, we need to reload the page.
```

Comment:
The reason being because updating `/auth/verify-email?rid=emailpassword&token=TOKEN` to `/auth/verify-email?rid=emailpassword` will not reload the component. 

There might be a way to leverage the `EmailVerification` feature state to reload the component while updating the URL too.

I'll try that.